minecraft

```diff
@@ -1,13 +1,27 @@
 #!/bin/sh
 
-# Move minecraft profile if detected at old location
-if [ -d $XDG_DATA_HOME/../.minecraft ]; then
-    if [ ! -f $XDG_DATA_HOME/minecraft/.migrated ]; then
-        echo "Migrating to XDG dirs"
-        mkdir -p $XDG_DATA_HOME/minecraft
-        cp -frl $XDG_DATA_HOME/../.minecraft/* $XDG_DATA_HOME/minecraft/
-        touch $XDG_DATA_HOME/minecraft/.migrated
+# A quick script to migrate back to the non-xdg directory
+# This is done due to the launcher not fully respecting the workdir argument so we have to hack our way around it.
+if [ -d $XDG_DATA_HOME/minecraft ]; then
+    if [ ! -f $XDG_DATA_HOME/../.minecraft/.nonxdg-migrated ]; then
+        echo "Migrating back to $XDG_DATA_HOME/../.minecraft"
+        
+        # Checks if there already is files in .minecraft.
+        # We move them to a safe folder in case someone for some reason stores something there.
+        # Version 2.2.1867 of the launcher actually also doesn't respect the workdir argument properly so this might just back up a log file.
+        if [ "$(ls -A $XDG_DATA_HOME/../.minecraft)" ]; then
+            echo "$XDG_DATA_HOME/../.minecraft not empty. Moving away files so that we don't overwrite anything..."
+            mkdir -p $XDG_DATA_HOME/../.minecraft-backup
+            mv $XDG_DATA_HOME/../.minecraft/* $XDG_DATA_HOME/../.minecraft-backup/
+        fi
+
+        mv $XDG_DATA_HOME/minecraft/* $XDG_DATA_HOME/../.minecraft/
+        rm -r $XDG_DATA_HOME/minecraft
+
+        # We create a symlink incase that we want to migrate back to using $XDG_DATA_HOME/minecraft
+        ln -s $XDG_DATA_HOME/../.minecraft $XDG_DATA_HOME/minecraft
+        touch $XDG_DATA_HOME/../.minecraft/.nonxdg-migrated
     fi
 fi
 
-exec /app/extra/minecraft-launcher/minecraft-launcher $@ --workDir=$XDG_DATA_HOME/minecraft
+exec /app/extra/minecraft-launcher/minecraft-launcher $@ -w $XDG_DATA_HOME/../.minecraft
```

Comment:
```suggestion
exec /app/extra/minecraft-launcher/minecraft-launcher $@ --workDir=$XDG_DATA_HOME/../.minecraft
```

`--workDir` is better documented and it is more likely that this argument will stay unchanged in the future than `-w`. ;-)

Reply:
I did this earlier but I actually noticed that CEF crashed the first time it initialized. However it did not crash with -w. 
I'll verify this again later.

Reply:
Hmm, this is strange.

Reply:
Go ahead and test it yourself if you have time to do so. Delete com.mojang.Minecraft and start the launcher (while using the --workDir argument)

Reply:
I just tried to reproduce this issue again and I couldn't. I guess it was just some weirdness with the launcher or a flaw from when I tested. Changed it to --workDir :)